language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java

```diff
@@ -196,6 +199,18 @@ private boolean withinLoopConstructs(T node) {
         return withinLoops;
     }
 
+    private boolean withinTransactionStatement(T node) {
+        Node evalNode = node;
+        boolean withinTransaction = false;
+
+        while (!withinTransaction && evalNode != null && evalNode.kind() != SyntaxKind.MODULE_PART) {
```

Comment:
Out of curiosity, can't we simplify this? Like removing the `MODULE_PART` check because the parent of a module part node is anyway `null`?

Reply:
Fixed